### DISCUSSION TITLE
All routes lost again - 4.1.0b2

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#567] [Q&A] [answered]
I'm done with nerves, what happened here? All my routes have been deleted by adding a single one and I have no backup. 
![Unbenannt](https://github.com/devemux86/cruiser/assets/121371629/34d72ca5-fd36-4258-9221-042bc668a933)

**Comment by DolomitesTom** [comment]
the same happened on my backup phone even if I didn't use it. This is a middle catastrophy.

**Reply by airald** [reply]
If you enabled to save your location on google services you can recover your tracks in google maps - my chronology. The data is limited, there are not timestamps as far I remember, but it's enough to recover your lost travel.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report and sorry about the case.

You are the only one reported this and unless we can reproduce it, it is difficult to guess what happens.
Files do not disappear by themselves.

Have you changed the app's default storage folder?
Where do you store them? In SD card? Media folder?

The last time you sent me your zip, it contained files with 0 size.

**Reply by DolomitesTom** [reply]
media folder internal, also these are empty . I had a backup from May, but already change a lot since then because of coming big tours

**Reply by devemux86** [reply]
SD cards may always be the reason, as they could fail, not be readable properly by Android, so apps do not see them.

But you mention internal storage? Even if it is the media, it should be more safe.

**Comment by DolomitesTom** [comment]
this must not happen that everything is deleted by a save or add action. Can you please check the routines again to see where this could happen?... also maybe by making routes visible or hidden on the map and saving afterwards?

**Reply by devemux86** [reply]
@DolomitesTom can you please remember your steps?
Did you save a new route, deleted a route, reorder routes or something else?

Also what exactly happened in the 2nd phone?

I will review the process to see how we can improve it. :slightly_smiling_face:

**Reply by devemux86** [reply]
@DolomitesTom how many routes?

Did you press save and exit the app?

**Comment by DolomitesTom** [comment]
I have no idea what happened on the second phone, there was also only 1 route left, all others have gone. I only recognized this now. I assume I did the same actions as on the 1st phone. 

I loaded a route from my routes. The  problem is that the folder is not remembered, I always on the first time when saving changes, have to select the right folder to overwrite with the same name. Can you change this that folder is also remembered, also when exiting the app in between?

I have then changed some Wps and saved with a different name, I also did sorting by name and pressed the save button, then exit button. I don't know any more in which sequence

**Reply by devemux86** [reply]
> I loaded a route from my routes. The problem is that the folder is not remembered, I always on the first time when saving changes, have to select the right folder to overwrite with the same name. Can you change this that folder is also remembered, also when exiting the app in between?

@DolomitesTom I will also see this.

**Comment by DolomitesTom** [comment]
Again all my routes have been deleted. It happened when I have created the new group, renamed the group and saved an imported route from a kurviger file in this new group. All routes of other groups have been deleted when saving the new route in the new group.

![Screenshot_20240702_132956_Cruiser](https://github.com/devemux86/cruiser/assets/121371629/ccc209c3-f7ff-4523-9011-0f2621b2da07)

**Reply by devemux86** [reply]
@DolomitesTom 

Save is done via a backup process to be sure and avoid any problems.

If you have any collection with routes, please send them with an email, so I can make tests with your structure.

Where do you save them, internal storage or SD card?
Data or media?

**Reply by DolomitesTom** [reply]
Routes are saved in the internal Android Media folder. I have try to simulate it, but I cannot reproduce it. I have no clue why and when it happens. But it happened already 3 times.

**Reply by devemux86** [reply]
@DolomitesTom 

> All routes of other groups have been deleted when saving the new route in the new group.

What do you mean "**saving** the new route in the new group" ?

- Pres the save button at bottom right?

- Or only import the route into the group without pressing the save button?

**Reply by DolomitesTom** [reply]
Long press on the route and then more... save route... I try to reproduce it, but couldn't. It must be some special situation, but I don't know which exactly.

**Reply by DolomitesTom** [reply]
What is saved then only this route or everything? When this happened really every route was deleted in the storage, is there maybe a problem with the numbering because it gets stored only with numbers not with names?

**Reply by DolomitesTom** [reply]
problem reading/writing array?

**Reply by devemux86** [reply]
@DolomitesTom 

Do you still have the "routes" folder after this happened?
Can you send it to me?

**Reply by DolomitesTom** [reply]
![Screenshot_20240703_125641_Total Commander](https://github.com/devemux86/cruiser/assets/121371629/e1b58acb-32d0-4b8b-a094-4f32ed72afb9)


The folders still exist, but they are empty. I have already restored them from an older archive.

Is it maybe a sorting problem of the numbers of the subfolders on the storage? The question is also why get routing Files deleted when I only want to add or change a new one? Something is wrong with the save changes routine (save button).

**Reply by devemux86** [reply]
@DolomitesTom 

Please send me your folders, filled + empty (if exist). This will help testing.

**Comment by devemux86** [comment, marked as answer]
This will be handled in v2 of the storage of favorites / routes / tracks:
- https://github.com/devemux86/cruiser/discussions/585
